### PR TITLE
CI: prepare for the new ironic container

### PR DESCRIPTION
**Pull request by dtantsur**
We will switch to using a combined API+conductor Ironic process in https://github.com/metal3-io/baremetal-operator/pull/1065.

### CODE REVIEW

**Comment by dtantsur** [comment]
/assign @furkatgofurov7

Could you take a careful look please?

File: vm-setup/roles/v1aX_integration_test/tasks/move.yml [hunk 1]
```diff
@@ -29,6 +30,9 @@
           
       - name: Fetch container logs before pivoting (kind cluster)
         shell: "sudo {{ CONTAINER_RUNTIME }} logs {{ item }} > /tmp/{{ CONTAINER_RUNTIME }}/{{ item }}/stdout.log 2> /tmp/{{ CONTAINER_RUNTIME }}/{{ item }}/stderr.log"
+        # FIXME(dtantsur): remove ignore_errors after the transition period
+        # from ironic-api+conductor to just ironic
```
Comment (furkatgofurov7) [inline vm-setup/roles/v1aX_integration_test/tasks/move.yml:34]: Am I getting it right here, that the end goal is to have only one container and get rid of ironic-api+conductor?

File: vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml [hunk 2]
```diff
@@ -268,6 +268,7 @@
   - name: Set expected ironic image based containers
     set_fact:
       ironic_image_containers:
+        - ironic
```
Comment (furkatgofurov7) [inline vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml:271]: Since this is touching an upgrade, we have to run tests that checks these changes

**Review by furkatgofurov7 — COMMENTED** [review]
Looks good, some questions are inline:

**Comment by furkatgofurov7** [comment]
@dtantsur one more thing I forgot to mention, since the new ironic container will be introduced in upgrade tests in this repo (ansible based), we have similar upgrade E2E tests (go based) in CAPM3, and I would expect to add this container in those tests too, to be specific to [this](https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/test/e2e/upgrade_ironic_test.go#L34-L47) list of containers.

**Comment by dtantsur** [comment]
Done: https://github.com/metal3-io/cluster-api-provider-metal3/pull/464

Now looking into the failing upgrade job.

**Review by furkatgofurov7 — APPROVED** [review]
/approve

/cc @lentzi90 @kashifest @Xenwar

**Comment by furkatgofurov7** [comment]
@dtantsur can you please update the PR description and mention/link the BMO PR to this so it is easier to understand the relation between this and BMO patch? Thanks.

**Review by lentzi90 — APPROVED** [review]
LGTM!

**Pull request merged**